quipucords/api/source/view.py

```diff
@@ -56,6 +60,105 @@ def format_source(json_source):
     return json_source
 
 
+@api_view(["post"])
+def source_bulk_delete(request):
+    """
+    Bulk delete sources.
+
+    Response payload contains IDs of sources deleted, skipped, and not found.
+    Example response:
+
+        {
+            "message": \
+                "Deleted 3 sources. "\
+                "Could not find 0 sources. "\
+                "Failed to delete 2 sources.",
+            "deleted": [1, 2, 3],
+            "missing": [],
+            "skipped": [
+                {"source": 6, "scans": [8], "scanjobs": [], "scantasks": []},
+                {
+                    "source": 7,
+                    "scans": [9, 10],
+                    "scanjobs": [11, 12],
+                    "scantasks": [13, 14, 15, 16],
+                },
+            ],
+        }
+
+    input:      "ids" : List of ids to delete, or string ALL_IDS_MAGIC_STRING
+    returns:    200 OK - upon successfully deleting any sources.
+                400 Bad Request - ids list is missing or empty.
+    """
+    ids = set_of_ids_or_all_str(request.data.get("ids"))
+
+    with transaction.atomic():
```

Comment:
I thought we were no longer going with atomic transactions for the new bulk deletes

Reply:
This transaction isn't strictly necessary in the general happy path, but I don't think it's hurting, and it might help in some narrow edge cases.

In both bulk delete APIs, we now explicitly search for and filter out any objects that we believe could prevent the delete from succeeding. However, Django doesn't _actually_ issue a single `DELETE` command to the database when you tell the model to delete a lot of objects. Internally, it still iterates and triggers signals related to the models, and it's possible that another request at the same time could cause a race condition in the related objects. It's admittedly a very narrow possibility for our app, but I believe that we probably _would_ want a transaction to protect and roll back in that case. Otherwise, we'd need to implement more explicit try-except logic in both view functions to prevent the response from halting and returning a `500` error.